engine/src/main/java/io/camunda/zeebe/engine/processing/message/MessageBatchExpireProcessor.java

```diff
@@ -27,12 +32,22 @@ public MessageBatchExpireProcessor(final StateWriter stateWriter) {
 
   @Override
   public void processRecord(final TypedRecord<MessageBatchRecord> record) {
-    record
-        .getValue()
-        .getMessageKeys()
-        .forEach(
-            messageKey ->
-                stateWriter.appendFollowUpEvent(
-                    messageKey, MessageIntent.EXPIRED, emptyDeleteMessageCommand));
+
+    int expiredMessagesCount = 0;
+    final int totalMessagesCount = record.getValue().getMessageKeys().size();
+    for (final long messageKey : record.getValue().getMessageKeys()) {
+      try {
+        stateWriter.appendFollowUpEvent(messageKey, EXPIRED, emptyDeleteMessageCommand);
+        expiredMessagesCount++;
+      } catch (final ExceededBatchRecordSizeException exceededBatchRecordSizeException) {
+        LOG.warn(
+            "Expected to expire messages in a batch, but exceeded the resulting batch size after expiring {} out of {} messages. "
+                + "Try using a lower Message TTL Checker's batch limit. (maxCommandsInBatch: {})",
+            expiredMessagesCount,
+            totalMessagesCount,
+            exceededBatchRecordSizeException);
```

Comment:
👍 I like how you've formulated the warning. This is helpful to users.

❌ There are three arguments expected, but only two are provided. The exception (i.e. the last parameter) is not considered an argument for the warning message template. No argument is provided for the `maxCommandsInBatch`.

🔧 Since we don't explicitly know how many commands fit in a batch (based on total byte size, not on the number of records), it may be difficult to provide a defined `maxCommandsInBatch`. Instead, you could consider appending the message of the caught exception. It contains more detailed information about the failure to append. However, that is perhaps too much internal information. Another idea would be to add the notion that the user could increase the `maxMessageSize`, although in most cases we recommend against this for performance, AFAIK. The size they would want to increase it to is the `currentBatchSize + entryLength`.

https://github.com/camunda/zeebe/blob/269dd99aaef112a04d2750771f65a26628d149a0/stream-platform/src/main/java/io/camunda/zeebe/stream/api/records/ExceededBatchRecordSizeException.java#L26-L27

Reply:
Thanks @korthout for your review. 

- I got some inspiration from this [error message](https://github.com/camunda/zeebe/blob/269dd99aaef112a04d2750771f65a26628d149a0/stream-platform/src/main/java/io/camunda/zeebe/stream/impl/ProcessingStateMachine.java#L96) which logs the `maxCommandsInBatch`. I also think this is an internal information and not necessarily useful for a self-healing warning. I would remove this part then if you agree.

- I'm not completely sure how to communicate to the user which parameters to change to get rid of the warning. `currentBatchSize + entryLength` are variable names and I don't think mentioning them is useful. Are there any examples of something similar like pointing to documentation or mentioning the config parameter?

Reply:
Sorry @abbasadel, I meant to say that the user could increase the `maxMessageSize` to the value of `batchSize + entryLength` (both of these are [passed to the constructor of the exception](https://github.com/camunda/zeebe/blob/269dd99aaef112a04d2750771f65a26628d149a0/stream-platform/src/main/java/io/camunda/zeebe/stream/api/records/ExceededBatchRecordSizeException.java#L21-L23C21)). But again, I think mentioning the maxmessagesize is not something we should do. Just informing the user to lower the Message TTL Checker's batch limit is best IMO.